src/events/l1_event_watcher.rs

```diff
@@ -91,3 +114,52 @@ where
 
     Ok(decoded_logs)
 }
+
+async fn fetch_deposit_hash_appended_events(
+    db_pool: &PgPool,
+    rpc_url: &str,
+    from_block: u64,
+    contract_addr: &str,
+) -> Result<Vec<Log<ZeroXBridge::DepositHashAppended>>, Box<dyn std::error::Error>> {
+    let event_name = ZeroXBridge::DepositHashAppended::SIGNATURE;
+    let logs = fetch_events_logs_at_address(rpc_url, from_block, contract_addr, event_name).await?;
+
+    for log in &logs {
+        // let event = &log.data;
+        //let event = log.data();
+        let event: &ZeroXBridge::DepositHashAppended = log.data();
+        let deposit_hash_event = DepositHashAppended {
+            id: 0, // Set by database
+            index: event.index.as_u64() as i64,
+            commitment_hash: event.commitmentHash.0.to_vec(),
+            root_hash: event.rootHash.0.to_vec(),
+            elements_count: event.elementsCount.as_u64() as i64,
+            block_number: log.block_number.ok_or("Block number not found")?.to::<u64>() as i64,
```

Comment:
It's much more more simpler to do this:

```suggestion
            block_number: log.block_number.ok_or("Block number not found")? as i64,
```